### ISSUE TITLE
Implement plotting methods for the model object

### ISSUE THREAD

**Issue by benRenard** [DassHydro/smash#286]
For instance:

* Obs vs. sim at some or all gauges, e.g. model.plot_obs_vs_sim(gauge=['ID1', 'ID2']) 
* simulated fields (either q or states) at specific time steps, e.g. model.plot_sim(field='q',time_step=["1998-05-23", "1998-05-24", "1998-05-25"])
* same for inputs, e.g. model.plot_input(field='prcp',time_step=["1998-05-23", "1998-05-24", "1998-05-25"])
* mesh properties, e.g. model.plot_mesh(attribute='flwdir')
* parameter fields, e.g. model.plot_par(par=['cp','ct'])
* and surely many others...

**Comment by nghi-truyen** [comment]
Yes, or we can even create a sub-package smash.plot to do this..
